File: TerraObserver/TO.FSharp/GodotAbstractions/Extensions/Planets/IPlanet.fs
namespace TO.FSharp.GodotAbstractions.Extensions.Planets

open TO.FSharp.GodotAbstractions.Bases

/// Copyright (C) 2025 Zhu Xiaohe(aka ZeromaXHe)
/// Author: Zhu XH (ZeromaXHe)
/// Date: 2025-05-28 23:04:28
[<Interface>]
type IPlanet =
    inherit INode3D
    abstract Radius: float32 with get
    abstract Divisions: int with get
    abstract ChunkDivisions: int with get
    abstract MaxHeight: float32 with get
    abstract StandardScale: float32 with get

File: TerraObserver/Godot.Abstractions/Extensions/Planets/IPlanet.cs
using Godot.Abstractions.Bases;

namespace Godot.Abstractions.Extensions.Planets;

/// Copyright (C) 2025 Zhu Xiaohe(aka ZeromaXHe)
/// Author: Zhu XH (ZeromaXHe)
/// Date: 2025-06-03 17:38:03
public interface IPlanet : IResource
{
    event Action? ParamsChanged;
    float Radius { get; set; }
    int Divisions { get; set; }
    int ChunkDivisions { get; set; }
    float UnitHeight { get; }
    float MaxHeight { get; }
    float MaxHeightRatio { get; }
    float StandardScale { get; }
    int ElevationStep { get; }
}